### PR TITLE
[realsense2] Update to 2.56.2

### LINKED ISSUE #38635: [realsense2] update to 2.55.1
### Library name

RealSense2

### New version number

2.55.1

### Other information that may be useful (release notes, etc...)

Release notes:
https://github.com/IntelRealSense/librealsense/releases/tag/v2.55.1

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Fixes #38635. Update `realsense2` to 2.56.2. 
Remove outdated patches `fix-osx.patch` and `${ARM64_DETECTION_FIX}`. 
Add and fix dependency `nlohmann-json`. 
Add `fix-android-prefix-path.patch` to fix android prefix path.

Feature `openni2` is tested successfully in the following triplets:
```
x86-windows
x64-windows
```
Feature `tools` is tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```
The usage test passed on `x64-windows` (header files found).

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/realsense2/fix-nlohmann_json.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+diff --git a/third-party/CMakeLists.txt b/third-party/CMakeLists.txt
+index c703140..cfb7d85 100644
+--- a/third-party/CMakeLists.txt
++++ b/third-party/CMakeLists.txt
+@@ -1,6 +1,7 @@
+ string(REPLACE ${PROJECT_SOURCE_DIR}/ "" _rel_path ${CMAKE_CURRENT_LIST_DIR})
+ 
+-include(CMake/external_json.cmake)
++find_package(nlohmann_json CONFIG REQUIRED)
++target_link_libraries(${LRS_TARGET} PRIVATE nlohmann_json::nlohmann_json)
```
Comment (dg0yt) [inline ports/realsense2/fix-nlohmann_json.patch:21]: Is this lib used by `${LRS_TARGET}`...

File: ports/realsense2/fix-nlohmann_json.patch [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+--- a/third-party/rsutils/CMakeLists.txt
++++ b/third-party/rsutils/CMakeLists.txt
+@@ -4,15 +4,15 @@ cmake_minimum_required(VERSION 3.8.0)  # source_group(TREE)
+ project( rsutils )
+ 
+ add_library( ${PROJECT_NAME} STATIC "" )
+-# We cannot directly interface with nlohmann_json (doesn't work on bionic)
+-#target_link_libraries( ${PROJECT_NAME} PUBLIC nlohmann_json )
++
++target_link_libraries(${PROJECT_NAME} PUBLIC nlohmann_json::nlohmann_json)
```
Comment (dg0yt) [inline ports/realsense2/fix-nlohmann_json.patch:36]: ... or by `${PROJECT_NAME}`?

**Comment by MonicaLiu0311** [comment]
Error on Android:
```
[221/222] cd /mnt/vcpkg-ci/b/realsense2/arm-neon-android-dbg && /android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar -M < /mnt/vcpkg-ci/b/realsense2/arm-neon-android-dbg/realsense2-all.ar
FAILED: /librealsense2-all.a 
cd /mnt/vcpkg-ci/b/realsense2/arm-neon-android-dbg && /android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar -M < /mnt/vcpkg-ci/b/realsense2/arm-neon-android-dbg/realsense2-all.ar
/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar: error: /librealsense2-all.a: Permission denied
ninja: build stopped: subcommand failed.
```

**Comment by dg0yt** [comment]
> error: /librealsense2-all.a: Permission denied

This seems to be lacking a prefix path and should be fixed as part of the update.

**Pull request merged**